### PR TITLE
Cleanup DataObject.FormatEnumerator and add DataObject FormatEnumerator tests

### PR DESCRIPTION
**Pull request by hughbe**
## Proposed Changes
- Cleanup DataObject.FormatEnumerator and add DataObject FormatEnumerator tests
- Remove dead CAS code

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/3189)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/DataObject.FormatEnumerator.cs [hunk 1]
```diff
@@ -20,65 +20,30 @@ public partial class DataObject
         /// </summary>
         private class FormatEnumerator : IEnumFORMATETC
         {
-            internal IDataObject parent = null;
-            internal ArrayList formats = new ArrayList();
-            internal int current = 0;
+            private IDataObject _parent;
+            private readonly ArrayList _formats = new ArrayList();
```
Comment (JeremyKuhne) [inline src/System.Windows.Forms/src/System/Windows/Forms/DataObject.FormatEnumerator.cs:24]: Can we make this `List<FORMATETC>`?

**Comment by JeremyKuhne** [comment]
Just one minor comment. It would also be good to call out in the description that you're removing dangling CAS related code to explain what is being removed.

---
In reply to: [412991048](https://github.com/dotnet/winforms/pull/3189#issuecomment-412991048) [](ancestors = 412991048)

File: src/System.Windows.Forms/src/System/Windows/Forms/DataObject.FormatEnumerator.cs (hunk 1 above)
Comment (hughbe) [reply src/System.Windows.Forms/src/System/Windows/Forms/DataObject.FormatEnumerator.cs]: Yes we can! Fixed

**Comment by hughbe** [comment]
Description updated

**Pull request merged**